### ISSUE TITLE
Universe bug

### ISSUE THREAD

**Issue by andricicezar** [FStarLang/FStar#2646]
This code makes F* to be in an incosistent state:
```fstar
module Test

noeq type monad = {
  m    : Type u#a -> Type u#(max 1 a);
  ret  : #a:Type -> a -> m a;
}

type mon_bind (mon:monad) = #a:Type u#a -> #b:Type u#b -> mon.m a -> (a -> mon.m b) -> mon.m b
```
The error reported by F*:

> (Error 18) ASSERTION FAILURE: Universe variable U_name a not found
> F* may be in an inconsistent state.
> Please file a bug report, ideally with a minimized version of the program that triggered the error.
>

**Comment by nikswamy** [comment]
I can't reproduce this with the latest 2022-07-13 release. I fixed issues similar to this one in PR #2630  

F* reports the `mon_bind` is incorrect, since `mon.m a` and `mon.m b` can both be correct (`a` and `b` are in different universes).

This works, but I suspect you really are trying to encode some kind of double universe polymorphic bind:

```
type mon_bind (mon:monad) = #a:Type u#a -> #b:Type u#a -> mon.m a -> (a -> mon.m b) -> mon.m b
```

**Comment by andricicezar** [comment]
Hello. Thanks for looking into this! Yeah, I was not on the latest release, sorry for that.

Yes, I am trying to encode a double universe polymorphic bind that I also use to define an indexed effect. I saw in the Universe chapter of the tutorial that it is not possible yet to do what I intend.
